### PR TITLE
Compute a proper flow digest for switch continuations

### PR DESCRIPTION
**Pull request by jcharum**
Compute a proper flow digest for switch expression evaluation continuations. The existing implementation was very incomplete and results in equivalent digests for non-equivalent expressions. The digest now includes the entire remainder of the switch expression to be evaluated.

Unit tests are added to verify.

This also contains a bit of refactoring to eliminate stutter ("evalSwitch.eval*").

### CODE REVIEW

**Comment by jcharum** [comment]
I did this in GitHub, because @mariusae reviewed the initial switch implementation and is still the world's leading expert on flow digests.  However, @swami-m, I think we should also seek to be world's leading experts.

File: syntax/switch.go [hunk 1]
```diff
@@ -83,24 +108,46 @@ func (c *CaseClause) Equal(d *CaseClause) bool {
...
+func (s *switcher) eval() (values.T, error) {
+	return s.evalCases(s.e.CaseClauses)
+}
+
+func (s *switcher) evalCases(cs []*CaseClause) (values.T, error) {
 	if len(cs) == 0 {
-		// This means no case clause matched.  This is an error, as the switch
-		// expression has no value.
-		return nil, fmt.Errorf("%s: no case pattern matches value", s.pos)
+		// Exhaustiveness-checking should prevent us from getting here.
```
Comment (mariusae) [inline syntax/switch.go]: in this case should we panic?

Comment (jcharum) [reply syntax/switch.go]: Done in [separate PR](https://github.com/grailbio/reflow/pull/123).

**Comment by jcharum** [comment]
@swami-m @prasadgopal: Could one of you have a look at this, as I think it would be good for someone at GRAIL to understand it?  As it stands, `switch` caching is pretty broken.

**Pull request merged**